### PR TITLE
8344235: Revisit SecurityManager usage in java.logging after JEP 486 and JEP 491 integration

### PR DESCRIPTION
**Pull request by dfuch**
This PR remove usage of SecurityManager, doPrivileges, etc... from `java.logging` and `java.base/jdk.internal.logger`

Only notable hack - Logger.checkPermission() no longer checks permissions, but has been renamed into `ensureLogManagerInitialized()` in order to avoid disturbing the initialization sequence of Logger/LogManager.

I am not 100% sure this is still needed - but I remember we had some entanglement issues in the past that had been hard to solve, so it seemed prudent to keep the call:

```
    if (manager == null) {
        manager = LogManager.getLogManager();
    }
```

where `manager` is a private volatile field in Logger.

I also took the opportunity to remove the locking workaround that had been introduced to support Virtual Threads and revert to using synchronized in the Handler classes now that JEP 491 has been integrated.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8344235](https://bugs.openjdk.org/browse/JDK-8344235): Revisit SecurityManager usage in java.logging after JEP 486 and JEP 491 integration (**Bug** - P4)

### Reviewers
 * [Jaikiran Pai](https://openjdk.org/census#jpai) (@jaikiran - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22281/head:pull/22281` \
`$ git checkout pull/22281`

Update a local copy of the PR: \
`$ git checkout pull/22281` \
`$ git pull https://git.openjdk.org/jdk.git pull/22281/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22281`

View PR using the GUI difftool: \
`$ git pr show -t 22281`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22281.diff">https://git.openjdk.org/jdk/pull/22281.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22281#issuecomment-2489411298)
</details>

### CODE REVIEW

File: src/java.base/share/classes/jdk/internal/logger/LoggerFinderLoader.java [hunk 1]
```diff
@@ -122,24 +107,13 @@ private static ErrorPolicy configurationErrorPolicy() {
...
-            ("jdk.logger.finder.singleton");
+        return Boolean.getBoolean("jdk.logger.finder.singleton");
     }
 
-    @SuppressWarnings("removal")
     private static Iterator<System.LoggerFinder> findLoggerFinderProviders() {
-        final Iterator<System.LoggerFinder> iterator;
-        if (System.getSecurityManager() == null) {
-            iterator = ServiceLoader.load(System.LoggerFinder.class,
+        final Iterator<System.LoggerFinder> iterator =
```
Comment (RogerRiggs) [inline src/java.base/share/classes/jdk/internal/logger/LoggerFinderLoader.java:114]: Could be simply be `return ` ... no need for a local var.

**Review by RogerRiggs — COMMENTED** [review]
Looks good

**Comment by dfuch** [comment]
> Hello Daniel, the changes look good to me. However, these changes appear to include both JEP 486 related updates as well as the JEP 491 (the reverting of ReentrantLock usage in favour of synchronized). Doing this clean up together I think is fine. However, I think we should update the issue title to also include JEP 491 in addition to JEP 486.

Oh right. I should have mentioned that in the summary. This is now done, I modified the JBS and PR titles as suggested too.

File: src/java.base/share/classes/jdk/internal/logger/LoggerFinderLoader.java (hunk 1 above)
Comment (dfuch) [reply src/java.base/share/classes/jdk/internal/logger/LoggerFinderLoader.java]: Done

**Review by jaikiran — APPROVED** [review]
This looks good to me.

File: src/java.base/share/classes/jdk/internal/logger/LazyLoggers.java [hunk 2]
```diff
@@ -350,10 +344,7 @@ private static LoggerFinder accessLoggerFinder() {
             // the result.
             // This is just an optimization to avoid the cost of calling
             // doPrivileged every time.
```
Comment (seanjmullan) [inline src/java.base/share/classes/jdk/internal/logger/LazyLoggers.java:346]: These 2 lines can be removed.

**Review by seanjmullan — COMMENTED** [review]
I noticed a comment that can be removed. Maybe next time you are in this file you can remember to remove it or you could fix it as part of another cleanup.

File: src/java.base/share/classes/jdk/internal/logger/LazyLoggers.java (hunk 2 above)
Comment (dfuch) [reply src/java.base/share/classes/jdk/internal/logger/LazyLoggers.java:346]: darn. I thought I had removed everything. Noted. thanks for noticing!

File: src/java.logging/share/classes/java/util/logging/LogManager.java [hunk 3]
```diff
@@ -461,29 +429,6 @@ private void readPrimordialConfiguration() { // must be called while holding con
     // Loggers are isolated from each AppContext.
     private LoggerContext getUserContext() {
         LoggerContext context = null;
-
-        @SuppressWarnings("removal")
-        SecurityManager sm = System.getSecurityManager();
-        JavaAWTAccess javaAwtAccess = SharedSecrets.getJavaAWTAccess();
```
Comment (turbanoff) [inline src/java.logging/share/classes/java/util/logging/LogManager.java:467]: `JavaAWTAccess` is now unused and can be removed

**Pull request closed without merging**